### PR TITLE
fix for #3281 Executing a node script through the RM restapi "node/script" on a busy node makes the task running forever

### PR DESCRIPTION
**Pull request by fviale**
- when executing a "node script", only terminate the script handler associated with the script execution, don't kill all active objects on the node
 - in ScriptExecutor, notify the manager only the node has been cleaned (to avoid asynchronous clean)
 - functional tests associated

### CODE REVIEW

File: scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        RMNodeEvent event;
+        do {
+            event = schedulerHelper.waitForAnyNodeEvent(RMEventType.NODE_STATE_CHANGED, TIMEOUT);
+        } while (!event.getNodeState().equals(NodeState.BUSY));
+
+        log("Wait when task starts");
+        schedulerHelper.waitForEventTaskRunning(jobId, "Test task");
+
+        log("Wait some time until task launcher is started");
+        Thread.sleep(5000);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                                                                                 TargetType.NODE_URL.name(),
+                                                                                 Sets.newSet(testNode.getNodeURL()));
+
+        log("Submit job");
+        final JobId jobId = schedulerHelper.submitJob(createNonBlockingJob(withSelectionScript));
+
+        // We wait a bit to be sure the job went through the scheduling loop
+        // if a selection script is set, the task start will be delayed as much as the SelectionManager is busy executing this script
+        // (and thus cannot execute the selection script provided by the task)
+        Thread.sleep(3000);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java [hunk 3]
```diff
@@ -567,23 +568,28 @@ private void checkContentAuthorization(String content) {
...
                 @Override
                 public ScriptResult<T> call() throws Exception {
+                    // scripts can only be executed on deployed nodes
+                    RMNodeImpl nodeImpl = (RMNodeImpl) node;
                     // Execute with a timeout the script by the remote handler
                     // and always async-unlock the node, exceptions will be
                     // treated as ExecutionException
                     try {
-                        ScriptResult<T> res = node.executeScript(script, bindings);
+                        logger.info("Executing node script on " + nodeImpl.getNodeURL());
```
Comment (sophiesongge) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java]: or getHostName()?

**Comment by sophiesongge** [comment]
We still have  rmnode.clean(); which will call node.killAllActiveObjects(); in ScriptExecutor.java

File: scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+        log("Check job result");
+        checkJobResult(schedulerHelper.getSchedulerInterface(), jobId);
+        checkActiveObjects();
+    }
+
+    private void checkActiveObjects() throws NodeException, ActiveObjectCreationException, InterruptedException {
+        try {
+            // This is to avoid slight asynchronous delays in active objects termination
+            Thread.sleep(1000);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java [hunk 5]
```diff
@@ -57,6 +57,7 @@
 import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
 import org.ow2.proactive.resourcemanager.exception.NotConnectedException;
 import org.ow2.proactive.resourcemanager.rmnode.RMNode;
+import org.ow2.proactive.resourcemanager.rmnode.RMNodeImpl;
```
Comment (sonarqube-activeeon) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java]: MINOR Remove this unused import 'org.ow2.proactive.resourcemanager.rmnode.RMNodeImpl'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by fviale** [comment]
Retest this please

**Pull request merged**